src/calendar.tsx

```diff
@@ -74,7 +75,10 @@ export function Calendar({
     }
 
     if (weekDateSelection) {
-      return onSelectDate({ start: startOfWeek(date), end: endOfWeek(date) })
+      return onSelectDate({
+        start: startOfWeek(date, { locale }),
```

Comment:
Do we also need to add `weekStartsOn` to the option here?

Reply:
yup, we'll need this for week's selection:

If we change the locale for `it` and the start of the week to Sunday and try to use the week selection, the start value will be monday (this is weird)

![image](https://user-images.githubusercontent.com/6943919/170253272-9cc44012-496c-44b3-9845-7370ff820ae2.png)